jetson/power_logging/model/save_model_summary.py

```diff
@@ -0,0 +1,97 @@
+import torch
+import json
+from typing import Any 
+
+def load_model(model_name: str, model_repo: str) -> Any:
+    """Load model from Pytorch Hub.
+
+    Args:
+        model_name: Name of model.
+            It should be same as that in Pytorch Hub.
+
+    Raises:
+        ValueError: If loading model fails from PyTorch Hub
+
+    Returns:
+        PyTorch model
+    """
+    try:
+        return torch.hub.load(model_repo, model_name, pretrained=True)
+    except:
+        raise ValueError(
+            f"Model name: {model_name} is most likely incorrect. "
+            "Please refer https://pytorch.org/hub/ to get model name."
+        )
+    
+
+def get_layers(model: torch.nn.Module, name_prefix: str="") -> list[tuple[str, torch.nn.Module]]:
+    """
+    Recursively get all layers in a pytorch model.
+
+    Args:
+        model: the pytorch model to look for layers.
+        name_prefix: Use to identify the parents layer. Defaults to "".
+
+    Returns:
+        a list of tuple containing the layer name and the layer.
+    """
+    children = list(model.named_children())
+
+    if len(children) == 0:
+        result = [(name_prefix, model)]
+    else:
+        result = []
+        for child_name, child in children:
+            layers = get_layers(child, name_prefix + "_" + child_name)
+            result.extend(layers)
+    
+    return result
+
+def get_layer_info(model, input_shape):
+    """
+    Get key information of all layers within a model.
+
+    Args:
+        model: The pytorch model
+        input_shape: input size of the model
+
+    Returns:
+        information about the model
+    """
+    model_info = {}
+    test = torch.randn(*input_shape)
+    hooks = []
+
+    def register_hook(layer_name):
+        def hook(module, input, output):
+            model_info[layer_name] = {
+                "input_shape": tuple(input[0].size()) if input else None,
+                "output_shape": tuple(output.size()) if output is not None else None,
+                "kernel_size": getattr(module, "kernel_size", None),
+                "stride": getattr(module, "stride", None),
+                "padding": getattr(module, "padding", None),
+                "type": module.__class__.__name__,
+            }
+        return hook
+
+    for layer_name, layer in get_layers(model):
+        hooks.append(layer.register_forward_hook(register_hook(layer_name)))
+    
+    model.eval()
+    with torch.no_grad():
+        _ = model(test)
+
+    for hook in hooks:
+        hook.remove()
+            
+    return model_info
+
+
+model = load_model("resnet18", "pytorch/vision:v0.10.0")
```

Comment:
Can be improved by adding all of these params to command line arguments (you can see an example of argparse usage in measure_inference_power.py